### ISSUE TITLE
Interactive Modal displays blank primary button

### ISSUE THREAD

**Issue by AnthonyOliveri** [carbon-design-system/carbon-components-react#139]
I created a Modal with the `secondaryButtonText` specified but not the `primaryButtonText`, like so:

```
<Modal
    modalHeading="Natural Language Classifier"
    modalLabel="PRICING PLANS"
    open={isServicePricingModalOpen}
    primaryButtonDisabled
    secondaryButtonText="Back"
    onRequestClose={goBack}
/>
```

The secondary button displays fine, but the primary button still shows with no text. If the `primaryButtonText` is not specified, I do expect it to not render at all.

![screen shot 2017-07-26 at 12 46 14 pm](https://user-images.githubusercontent.com/5459406/28635840-a639aec4-7201-11e7-83ae-5af4b3350cfd.png)

**Comment by AnthonyOliveri** [comment]
The same thing happens if I only specify the `primaryButtonText` but do not specify the `secondaryButtonText`.

![screen shot 2017-07-26 at 12 57 35 pm](https://user-images.githubusercontent.com/5459406/28635972-3aaad4a2-7202-11e7-8f96-ed48494ada04.png)

**Comment by hellobrian** [comment]
Primary and Secondary Buttons will always get rendered in a Modal.
The `primaryButtonDisabled` prop sets the `disabled` attribute to true making it unclick-able but still visible because we have [`disabled` styles for these buttons](http://react.carbondesignsystem.com/?selectedKind=Buttons&selectedStory=Disabled%20Button&full=0&down=1&left=1&panelRight=0&downPanel=storybook%2Factions%2Factions-panel).

```jsx
...
<Button
  kind="primary"
  disabled={primaryButtonDisabled}
  onClick={onRequestSubmit}
>
  {primaryButtonText}
</Button>
...
```

**Solution:** Since you're just trying to give users the ability to close the modal, rely on the X button in the top-right corner. Finally, you can hide the buttons by using a **passive Modal**, just pass in this prop:

```jsx
<Modal passiveModal>
  ...
</Modal>
```

For more details on using **Modals in React**, checkout our [react storybook here](http://react.carbondesignsystem.com/?selectedKind=Modal&selectedStory=passive&full=0&down=1&left=1&panelRight=0&downPanel=storybook%2Factions%2Factions-panel)

**Comment by AnthonyOliveri** [comment]
I understand the how disabled styling and passive modals work. I think I wasn't completely clear with why I opened this issue. The thing I don't understand is why the non-passive Modal type is constrained to exactly 2 buttons. Using the `X` button in the top right corner does not work in my case because I do not want to dismiss the modal, but go back to the previous modal. That's why I tried putting a single `Back` button. For this Modal, there is nothing to create or go forward to, so it makes no sense to have a 2nd button.

**Comment by hellobrian** [comment]
Typically Modals are constrained to exactly 2 buttons because most scenarios call for a binary response: yes/no, cancel/submit, and in your case, back/forward. Although this isn't the solution you want, the solution you have right now is that you're stuck with 2 or 0 buttons so, if you're working with 2 buttons, you should disable the appropriate button if you don't want your users to click on it.

But more importantly, I have concerns about the UX you're implementing:

> Using the X button in the top right corner does not work in my case because I do not want to dismiss the modal, but go back to the previous Modal. 

Based on this short description alone, it sounds like you have a Modal navigating to another Modal and that to me is not a good idea. 

But lastly, I personally think it could be better if our Modal was more composable, which would solve your situation and possibly other situations we haven't encountered yet. I can bring this up with my designers and feds and get their thoughts.

**Comment by hellobrian** [comment]
I'm talking with my team right now, we need more context because we're making a lot of assumptions about what you need based on the designs you're implementing. Can you share a screenshot of what you're trying to do with the Modal?

**Comment by AnthonyOliveri** [comment]
Thanks for the feedback. I'm fairly new to UI development, so I was unaware that Modals are usually constrained have exactly 2 buttons. I'll let my team know about this to see if can improve our implementation. Either way, it would be greatly appreciated if the buttons were more configurable.

As for the design, this is what my team currently has for the Watson developer console. There are 3 modals that let the user navigate from one to the next (or go back if they change their mind). For the last modal, clicking the `X` in the top right corner goes back to the previous modal. That is unintuitive, but it's the behavior we want (since they are dismissing a modal generated by the previous modal).

<img width="1250" alt="screen shot 2017-07-28 at 11 37 58 am" src="https://user-images.githubusercontent.com/5459406/28727202-9e98d4fe-7389-11e7-891f-55efeeb6adfb.png">

<img width="1061" alt="screen shot 2017-07-28 at 11 38 16 am" src="https://user-images.githubusercontent.com/5459406/28727207-a2493206-7389-11e7-8afa-94252ef57f39.png">

<img width="1031" alt="screen shot 2017-07-28 at 11 38 30 am" src="https://user-images.githubusercontent.com/5459406/28727211-a6c4e15e-7389-11e7-85c0-9a13301b27f3.png">

**Comment by chrisdhanaraj** [comment]
Just jumping in, you're completely right that the Modal component should be more composable. Think this was an artifact of an older style of component, I can add this to our next sprint to tackle

**Comment by joshblack** [comment]
Should be closed with the work in #223, right @marijohannessen?
